src/PostgREST/App.hs

```diff
@@ -197,9 +197,10 @@ app dbStructure proc conf apiRequest =
               else do
                 let r = contentRangeH 0 (toInteger $ queryTotal-1)
                           (toInteger <$> if shouldCount then Just queryTotal else Nothing)
-                    s = if iPreferRepresentation apiRequest == Full
+                    s_200 = if iPreferRepresentation apiRequest == Full
                           then status200
                           else status204
+                    s = if queryTotal == 0 then status404 else s_200
```

Comment:
Is this related to https://github.com/PostgREST/postgrest/issues/1257?

Reply:
Oh. That was a mistake, wrong branch. Sorry about that.

Yes. I was trying to fix it, but my current approach, breaks a few other tests.

I can create a PR and we can discuss it there.

Reply:
Thank you. That makes reviewing easier.